core/templates/dev/head/domain/utilities/UrlInterpolationServiceSpec.js

```diff
@@ -0,0 +1,114 @@
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for UrlInterpolationService.
+ *
+ * @author henning.benmax@gmail.com (Ben Henning)
+ */
+
+describe('URL Interpolation Service', function() {
+  var uis = null;
+
+  beforeEach(module('oppia'));
+
+  beforeEach(inject(function($injector) {
+    uis = $injector.get('UrlInterpolationService');
+  }));
+
+  it('should return the same URL for erroneous URLs', function() {
+    expect(uis.interpolateUrl()).toBe(undefined);
+    expect(uis.interpolateUrl(null, {})).toBe(null);
+    expect(uis.interpolateUrl(undefined, {})).toBe(undefined);
+    expect(uis.interpolateUrl('', {})).toBe('');
+    expect(uis.interpolateUrl('')).toBe('');
+  });
+
+  it('should return null for erroneous interpolation values', function() {
+    expect(uis.interpolateUrl('url', null)).toBe(null);
+    expect(uis.interpolateUrl('url', undefined)).toBe(null);
+    expect(uis.interpolateUrl('/test_url/<param>', 'value')).toBe(null);
+    expect(uis.interpolateUrl('/test_url/<param>', ['value'])).toBe(null);
+  });
+
+  it('should interpolate URLs not requiring parameters', function() {
+    expect(uis.interpolateUrl('/test_url/', {})).toBe('/test_url/');
+    expect(uis.interpolateUrl('/test_url/', {'param': 'value'})).toBe(
+      '/test_url/');
+  });
+
+  it('should interpolate URLs requiring one or more parameters', function() {
+    expect(uis.interpolateUrl('/test_url/<fparam>', {'fparam': 'value'})).toBe(
+      '/test_url/value');
+    expect(uis.interpolateUrl(
+      '/test_url/<first_param>/<second_param>/<third_param>', {
+        'first_param': 'value1',
+        'second_param': 'value2',
+        'third_param': 'value3'
+      })).toBe('/test_url/value1/value2/value3');
+  });
+
+  it('should interpolate parameters within words or adjacent to other parameters', function() {
+    // It also doesn't need to have '/' prefixing the URL.
+    expect(uis.interpolateUrl(
+      'word<with_param>', {'with_param': '_with_value'}))
+    .toBe('word_with_value');
+    expect(uis.interpolateUrl('Eating_<param1><param2>_with_syrup', {
+      'param1': 'pan', 'param2': 'cakes'
+    })).toBe('Eating_pancakes_with_syrup');
+  });
+
+  it('should interpolate parameters beginning, ending, or composing the URL', function() {
+    expect(uis.interpolateUrl('<prefix>_with_words', {'prefix': 'Signs'})).toBe(
+      'Signs_with_words');
+    expect(uis.interpolateUrl('Neither_here_nor_<suffix>', {
+      'suffix': 'anywhere'
+    })).toBe('Neither_here_nor_anywhere');
+    expect(uis.interpolateUrl('<param>', {'param': 'value'})).toBe('value');
+  });
+
+  it('should sanitize parameters but not URLs', function() {
+    expect(uis.interpolateUrl('URL with a space', {})).toBe(
+      'URL with a space');
+    expect(uis.interpolateUrl(
+      '/test_url/<first_param>?<query_name>=<query_value>', {
+        'first_param': 'SEARCH',
+        'query_name': 'title or website',
+        'query_value': 'http://oppia.org/'
```

Comment:
Really small nit: https://www.oppia.org. (I occasionally scan the codebase for http urls; we should always use https.)